src/ng/http.js

```diff
@@ -1249,12 +1260,19 @@ function $HttpProvider() {
           cache,
           cachedResp,
           reqHeaders = config.headers,
-          url = buildUrl(config.url, config.paramSerializer(config.params));
+          url = config.url;
+
+      if (lowercase(config.method) === 'jsonp') {
+        // This is a pretty sensitive operation where we're allowing a script to have full access to
+        // our DOM and JS space.  So we require that the URL satisfies SCE.RESOURCE_URL.
+        url = $sce.getTrustedResourceUrl(url);
```

Comment:
So, if I understand correctly, `config.url` must be a string for all non-JSONP requests (i.e. it can't be an `$sce`-wrapped value - not that it needs to). And for JSONP requests it can be either an `$sce`-trusted resourceUrl or matched by a rule in the resourceUrlWhiteList (but not in the blackList). Right?

It might be more staight-forward if we allowed wrapped values for other methods too. E.g.:

```js
if (... === 'jsonp') {
  ...
} else {
  url = $sce.valueOf(url);
}
```